controllers/runner_controller.go

```diff
@@ -398,10 +397,16 @@ func (r *RunnerReconciler) newPod(runner v1alpha1.Runner) (corev1.Pod, error) {
 		},
 	}
 
+	if runner.Spec.PodTemplate.ObjectMeta.Size() > 0 {
+		runner.Spec.PodTemplate.ObjectMeta.DeepCopyInto(&pod.ObjectMeta)
+	}
+
+	if runner.Spec.PodTemplate.Template.Spec.Size() > 0 {
+		runner.Spec.PodTemplate.Template.Spec.DeepCopyInto(&pod.Spec)
+	}
```

Comment:
I think it's better to first set `PodTemplate.Spec` to `pod.Spec` and then override the specific settings as follows:

```
podSpec := Spec: runner.Spec.PodTemplate.Template.Spec.DeepCopy()
pod := corev1.Pod{
  Spec: *podSpec, 
}

pod.Spec.RestartPolicy = "OnFailure"
...
```

However, how to override the `pod.Spec.Containers` field is still open for consideration...

Reply:
[As similar as K8s already do in strategic-merge patch, we can merge on `name ` for containers](https://github.com/kubernetes/client-go/blob/7ac12361944b3b98074594cf00977861b9b41e4b/pkg/api/v1/types.go#L2119).

// Too bad we can't reuse the strategic-merge itself in our case, as it seems like for generating patches, rather than merging objects :(

Reply:
@mumoshu, I missed the fact that @summerwind posted in the Issuehttps://github.com/summerwind/actions-runner-controller/issues/5#issuecomment-599968643. So seems like I will rewrite code to use specific fields of podspec

Reply:
According to the discussion made at #5, can we just defer implementing merge on `containers`?

Reply:
yeah. I will just go through fields, and assign them if not empty to current pod creation or merge additional fields like env :)